libs/martech/helpers.js

```diff
@@ -358,70 +339,179 @@ function getUrl() {
   if (host.includes('stage.adobe')
     || host.includes('corp.adobe')
     || host.includes('graybox.adobe')) {
-    return 'https://www.stage.adobe.com/experienceedge/v2/interact';
+    return `https://www.stage.adobe.com/experienceedge/v2/${isCollectCall ? 'collect' : 'interact'}`;
   }
 
   const { origin } = window.location;
-  return `${origin}/experienceedge/v2/interact`;
+  return `${origin}/experienceedge/v2/${isCollectCall ? 'collect' : 'interact'}`;
 }
 
-/**
- * Loads analytics and interaction data based on the user and page context.
- * Sends the data to Adobe Analytics and Adobe Target for personalization.
- *
- * @param {Object} params - The parameters for the function.
- * @param {Object} params.locale - The locale object containing language/region info.
- * @param {string} params.env - The environment (e.g., 'prod' for production).
- * @param {string} [params.calculatedTimeout] - timeout value for the request in milliseconds.
- *
- * @returns {Promise<Object>} A promise that resolves to the
- * personalization propositions fetched from Adobe Target.
- */
-export const loadAnalyticsAndInteractionData = async ({ locale, env, calculatedTimeout }) => {
+export const createRequestUrl = ({
+  env,
+  hitType,
+}) => {
+  const TARGET_API_URL = getUrl(hitType === 'propositionDisplay');
+  DATA_STREAM_ID = env === 'prod' ? '913eac4d-900b-45e8-9ee7-306216765cd2' : 'e065836d-be57-47ef-b8d1-999e1657e8fd';
+  if (hitType === 'pageView' || hitType === 'propositionDisplay') {
+    const isFirstVisit = !getCookie(AMCV_COOKIE);
+    const consentCookie = getCookie('OptanonConsent') || '';
+    if (isFirstVisit || !consentCookie.includes('C0004')) {
+      DATA_STREAM_ID = env === 'prod' ? '57c20bab-94c3-425e-95cb-0b9948b1fdd4' : 'a44f0037-2ada-441f-a012-243832ce5ff9';
```

Comment:
nit - better to assign IDs to descriptive consts

Also all caps should be used for actual constants, not derived.